### DISCUSSION TITLE
Can not select Svg GShapePreRenderedElement

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#922] [Q&A] [answered]
Hi,
    I want to use a svg as part of my node, one strange thing is that when I click the SVG element I can not select my node, Does anyone knows the magic of  SVG GShapePreRenderedElement?
mygraph code is 
 protected DemoGraph(): GGraph {
        const code = fs.readFileSync(join(__dirname, '../../src/images/SVG_logo.svg'), 'utf-8');
        return GGraph.builder()
            .id('root')
            .add(
                GNode.builder()
                    .id('node1')
                    .addCssClass('tasklist-node')
                    .layout('hbox')
                    .position(100, 100)
                    .add(GPort.builder().id('port1').size(10, 10).build())
                    .add(GCompartment.builder().id('Compartment').type('centerPart').size(120, 120).addCssClass('center-part').build())
                    .add(GPort.builder().id('port2').size(10, 10).build())
                    .build()
            )
            .add(
                GNode.builder()
                    .id('svg-node')
                    .addCssClass('tasklist-node')
                    .layout('hbox')
                    .position(300, 100)
                    .add(GPort.builder().id('port3').size(10, 10).build())
                    .add(
                        GShapePreRenderedElement.builder()
                            .id('svg')
                            .type('shape-pre-rendered')
                            .addCssClass('chip-projection')
                            .code(code)
                            .size(120, 120)
                            .build()
                    )
                    .add(GPort.builder().id('port4').size(10, 10).build())
                    .build()
            )
            .build();
    }
configureModelElement(context, 'centerPart', SCompartment, RectangularNodeView);
.tasklist-node .center-part {
    fill: #f1bd6c;
}
.chip-projection {
    background-color: rgba(204, 114, 38, 0.5);
}
![image](https://user-images.githubusercontent.com/4537809/218973554-0d71e00d-42b3-45cc-a493-5ed208493887.png)

![2023-02-15_15-56-35](https://user-images.githubusercontent.com/4537809/218967787-0c0c7b83-dc96-4207-b1ba-6d4f3fad7d84.gif)

**Comment by tortmayr** [comment, marked as answer]
Hi @wss29,
I guess the issue is that the `GShapePreRenderedElement` element itself is also selectable per default. 
So whenever you are clicking inside the node you are actually selecting the SVG logo and not the node itself. 
Since the logo doesn't have a CSS style for selection nothing happens visually.

You can disable the selection feature of the your prerendered elements like this:
```ts
    configureModelElement(context, 'shape-pre-rendered', PreRenderedElement, PreRenderedView, { disable: [selectFeature] });
```

**Reply by wss29** [reply]
Yes, you are very smart, It is the solution,Thank your very much
